typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts

```diff
@@ -0,0 +1,334 @@
+/*-
+ * #%L
+ * vrotsc
+ * %%
+ * Copyright (C) 2023 - 2024 VMware
+ * %%
+ * Build Tools for VMware Aria
+ * Copyright 2023 VMware, Inc.
+ *
+ * This product is licensed to you under the BSD-2 license (the "License"). You may not use this product except in compliance with the BSD-2 License.
+ *
+ * This product may include a number of subcomponents with separate copyright notices and license terms. Your use of these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE file.
+ * #L%
+ */
+
+/**
+ * The node holds information about it's position
+ */
+export interface GraphNode {
+	name: string;
+	x?: number;
+	y?: number;
+	targets: string[];
+}
+
+/**
+ * Leaf holds the node and it's children(leaves)
+ */
+export interface Leaf {
+	node: GraphNode;
+	leaves: Leaf[];
+}
+
+/**
+ * Tree holds the structure of the graph, it's populated with the calculated positions
+ */
+interface Tree {
+	startNodes: string[];
+	leaves: {
+		[nodeName: string]: Leaf;
+	};
+}
+
+/**
+ * This is used to chart out a graph and give you the x and y positions of each node
+ *
+ * Example:
+ * ```typescript
+const nodes = [
+	// First Start
+	{ name: "A", targets: ["B"] },
+	{ name: "B", targets: ["C"] },
+	{ name: "C", targets: ["D", "G"] },
+	{ name: "D", targets: ["E"] },
+	{ name: "E", targets: ["C", "F"] },
+	{ name: "F", targets: ["O"] },
+	{ name: "G", targets: ["H"] },
+	{ name: "H", targets: ["I"] },
+	{ name: "I", targets: ["J", "K", "L", "M"] },
+	{ name: "J", targets: [] },
+	{ name: "K", targets: [] },
+	{ name: "L", targets: [] },
+	{ name: "M", targets: [] },
+	{ name: "O", targets: ["P"] },
+	{ name: "P", targets: ["Q"] },
+	{ name: "Q", targets: [] },
+
+	// Second start
+	{ name: "S", targets: ["T"] },
+	{ name: "T", targets: ["U", "W", "D"] },
+	{ name: "U", targets: [] },
+	{ name: "W", targets: [] },
+
+	// Third Start
+	{ name: "X", targets: ["Y"] },
+	{ name: "Y", targets: [] },
+];
+
+const graph = new Graph(nodes, ["A", "S", "X"]);
+
+graph.draw();
+ * ```
+ *
+ * Will output:
+ * ```
+....................................................................................................
+....................................................................................................
+....................................................................................................
+..............................F....O....P....Q......................................................
+....................................................................................................
+....................................................................................................
+....................D....E..........................................................................
+....................................................................................................
+.....A....B....C...................J................................................................
```

Comment:
The placement of 'J' with root 'I' is confusing, I think it needs to be in the row below

Reply:
The calculations are not perfect here since they go down to fraction... rounding them up leads to deviations... look at the example for  C and D,G it's reversed... vRO somewhat supports this tho, so we are all good ... even tho vRO has it's own problems and it converts the calculations for x and y for some reason..... so it's not perfect, but it will do